[lihaoquan] iP

lihaoquan opened the pull request:
# Bob
Bob helps you to remember what to do. It's,

- text-based
- easy to learn
- fast to use

All you need to do is,

1. download it from [here](https://github.com/lihaoquan/ip/releases/tag/v0.1).
2. run the JAR file in a command line.
3. add your tasks.
4. let it manage your tasks for you.

And it is **FREE!**

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you are a Java programmer, you can make something better than this:
```java
package bob;

/**
 * Parse user input into commands.
 */
public class BobParser {

    public static final String TERMINATE_COMMAND = "bye";

    public static final String LIST_COMMAND = "list";
    public static final String MARK_COMMAND = "mark";
    public static final String UNMARK_COMMAND = "unmark";

    public static final String TODO_COMMAND = "todo";
    public static final String DEADLINE_COMMAND = "deadline";
    public static final String EVENT_COMMAND = "event";
    public static final String DELETE_COMMAND = "delete";
    public static final String FIND_COMMAND = "find";
```

> "I should've used enums." - Me (today)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

lihaoquan committed:
Update gitignore for Java & Added Level-0 Features

lihaoquan committed:
Customized Chatbot Personality & Added Echo Functionality

lihaoquan committed:
Add Storage to Store User Texts

lihaoquan committed:
Ability to Mark and Unmark Tasks

lihaoquan committed:
Small Modification to Tag as A-Classes

lihaoquan committed:
Add Support for Three Task Types

lihaoquan committed:
Automated Text UI Testing

lihaoquan committed:
Error Handling and Refactoring of Bob.java

lihaoquan committed:
Added Support for Deletion of Tasks

lihaoquan committed:
Revert README.md After Mistakenly Editing It

lihaoquan committed:
Add ability to save and modify tasks and refactor variables

lihaoquan committed:
Revert "Add ability to save and modify tasks and refactor variables"

This reverts commit dafc5870b793fce66537034f64a1f109f3944b67.

lihaoquan committed:
Small modification to reattempt branch merge

lihaoquan committed:
Merge branch 'branch-Level-7'

lihaoquan committed:
Add checkstyle.jar and 2103.xml config to fix styling errors

lihaoquan committed:
Import AB3 checkstyle xml config

lihaoquan committed:
Add datetime recognition to Deadline command

lihaoquan committed:
Merge branch 'branch-Level-8'

lihaoquan committed:
Extract Ui class from Bob.java

lihaoquan committed:
Extract Storage class from Bob.java

lihaoquan committed:
Extract Parser from Bob.java

lihaoquan committed:
Extract TaskList from Bob.java

lihaoquan committed:
Update test file to cater for directory changes

lihaoquan committed:
Put all classes into Bob package

lihaoquan committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

lihaoquan committed:
Automate project builds using Gradle

lihaoquan committed:
Add JUnit tests and patch Deadline display format

lihaoquan committed:
Package the App as a JAR file

lihaoquan committed:
Add JavaDoc comments to the code

lihaoquan committed:
Give users a way to find a task by searching for a keyword

lihaoquan committed:
Revert "Add JavaDoc comments to the code"

This reverts commit 183736a9d7c78584ed5d8c303f2b5363a9b3dd41.

lihaoquan committed:
Update BobTaskList.java due to mistake in parallel branches

lihaoquan committed:
Commit on purpose to introduce merge conflict

lihaoquan committed:
Merge branch 'branch-A-JavaDoc'

lihaoquan committed:
Merge branch 'branch-Level-9'

lihaoquan committed:
Refactor BobUI and add more Javadocs

lihaoquan committed:
Merge branch 'branch-A-JavaDoc'

lihaoquan committed:
Rename constants to comply with standards

lihaoquan committed:
Resolve JavaDoc merge conflicts

lihaoquan committed:
Merge branch 'branch-A-JavaDoc'

lihaoquan committed:
Update runtest.bat for package statement

lihaoquan committed:
Prepare for JavaFX integration

lihaoquan committed:
Remove code catering to command line

lihaoquan committed:
Rename Duke to Bob

ejnan commented on src/main/java/bob/Bob.java:
```diff
@@ -0,0 +1,74 @@
+     *
+     * @returns the Bot class object.
+     */
+    public static Bob init() {
```
Perhaps you can use a more descriptive name instead of init()? Maybe `createBobInstance()` or `initialiseBob()` would be clearer.

ejnan commented on src/main/java/bob/Task.java:
```diff
@@ -0,0 +1,57 @@
+public class Task {
+
+    protected String description;
+    protected boolean done = false;
```
According to our coding standard, Boolean variables/methods should be named to sound like booleans.
Maybe `isDone` would be better?

ejnan commented on src/main/java/bob/BobStorage.java:
```diff
@@ -0,0 +1,162 @@
+            try (BufferedReader br = new BufferedReader(new FileReader(this.SAVE_DATA))) {
+
+                String line;
+                boolean invalidFile = false;
```
According to our coding standard, it is recommended as to use a prefix such as _is, has, was,_ etc. for boolean variable/method names.
Maybe `isValidFile` is a better name for this variable?

ejnan reviewed (commented):
I found your code easy to read, with descriptive method names that also follow the coding standard format. I liked that the JavaDoc comments were also very clear and concise.
I noted a few minor coding standard violations, but other than that LGTM!

purivirakarin commented on src/main/java/bob/MainWindow.java:
```diff
@@ -0,0 +1,53 @@
+                DialogBox.getBobDialog("Hello, I'm Bob, a personal assistant. How can I help you?", bobImage));
+    }
+
+    public void setBob(Bob d) {
+        this.bob = d;
+    }
```
Can edit `d` into a more descriptive name

purivirakarin commented on src/main/java/bob/BobParser.java:
```diff
@@ -0,0 +1,68 @@
+            case BobParser.DELETE_COMMAND:
+                return this.taskList.handleTaskDeletion(input);
+            case BobParser.FIND_COMMAND:
+                return this.taskList.handleFindTask(input);
```
According to the coding standard, we may avoid the unnecessary use of `this` with fields.

```
// java Use the this keyword only when a field is shadowed by a method or constructor parameter.

// Good
public User(String name) {
    this.name = name;
    ...
}

// Bad
public User(String name) {
    // 'id' is not shadowed by any method parameters
    this.id = User.getNewId();
    ...
}

```

purivirakarin commented on src/main/java/bob/BobTaskList.java:
```diff
@@ -0,0 +1,218 @@
+
+                input = input.substring(BobParser.DEADLINE_COMMAND.length() + 1);
+
+                String[] split = input.split("/by");
```
According to the coding standard,

> Plural form should be used on names representing a collection of objects.

you may rename `split` into `splits`.

purivirakarin commented on src/main/java/bob/BobTaskList.java:
```diff
@@ -0,0 +1,218 @@
+    }
+
+    public ArrayList<Task> getList() {
+        return this.list;
```
Referring to the comment regarding the unnecessary of of `this`, you may omit `this.` here.

purivirakarin reviewed (commented):
Overall the code looks good to me. Most parts of the code follows the coding standard well and have descriptive comments and JavaDocs. Just some place I noted out about an unnecessary use of `this` can be avoided. In addition can pay attention to the field that represent a collection of objects, which may better be rename into a plural form.

lihaoquan committed:
Update build.gradle to create JAR file

Application's main class has been set to Launcher.

A JAR file is created to include JavaFX UI.

lihaoquan committed:
Add assert statements and refactor code

Assert statements are added as part of A-Assertions.

Some parts of the code are refactored to remove redundant text UI functions.

lihaoquan committed:
Update BobStorage.java

Add additional assert statements to
capture other possible errors with file
creation.

lihaoquan committed:
Partial refactor of code

Refactor a portion of the code.
Other files are work in progress.

lihaoquan committed:
Refactor all files for A-CodeQuality

All files are refactored to comply with code quality.

lihaoquan committed:
Merge pull request #1 from lihaoquan/branch-A-CodeQuality

branch-A-CodeQuality Pull Request

lihaoquan committed:
Merge branch 'master' into branch-A-Assertions

lihaoquan committed:
Merge pull request #2 from lihaoquan/branch-A-Assertions

branch-A-Assertions Pull Request

lihaoquan committed:
Update BobTaskList.java

Refactor file conditional checking to make it more readable.

Fix bugs introduced due to not trimming input.

lihaoquan committed:
Merge pull request #3 from lihaoquan/branch-A-CodeQuality

Update BobTaskList.java

lihaoquan committed:
Create gradle.yml

File used to set up Continuous Integration (CI)

lihaoquan committed:
Merge pull request #4 from lihaoquan/branch-A-CI

Create gradle.yml

lihaoquan committed:
Add ability to sort task by alphabetical order

Task can be sort alphabetically in either
ascending or descending order.

lihaoquan committed:
Add JavaDocs and refactor files

Added missing JavaDocs from running checkstyle.

Files have been refactored and new bugs were introduced.
These bugs are addressed and fixed.

lihaoquan committed:
Merge pull request #5 from lihaoquan/branch-C-Sort

Branch for C-SORT

lihaoquan committed:
Change BCD-Extension to C-Archive due to conflict

A teammate has done C-Sort already.
Changed task to C-Archive instead to satisfy BCD-Extension.

lihaoquan committed:
Add user guide and remove redundant code

User guide has been added as part of requirements for A-UserGuide.

Some code are also redundant, and are removed.

lihaoquan committed:
Update README.md

Remove comments and format texts.

lihaoquan committed:
Update README.md

Further formatting for text line breaks.
